test/e2e/e2e.go

```diff
@@ -0,0 +1,243 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	sourcesv1alpha1 "github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	servingV1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	rbacV1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	// Mysteriously required to support GCP auth (required by k8s libs).
+	// Apparently just importing it is enough. @_@ side effects @_@.
+	// https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	defaultNamespaceName = "e2etestfn3"
+	testNamespace        = "e2etest"
+	interval             = 1 * time.Second
+	timeout              = 1 * time.Minute
+)
+
+// Setup creates the client objects needed in the e2e tests.
+func Setup(t *testing.T, logger *logging.BaseLogger) (*test.Clients, *test.Cleaner) {
+	if pkgTest.Flags.Namespace == "" {
+		pkgTest.Flags.Namespace = defaultNamespaceName
+	}
+
+	clients, err := test.NewClients(
+		pkgTest.Flags.Kubeconfig,
+		pkgTest.Flags.Cluster,
+		pkgTest.Flags.Namespace)
+	if err != nil {
+		t.Fatalf("Couldn't initialize clients: %v", err)
+	}
+	cleaner := test.NewCleaner(logger, clients.Dynamic)
+
+	return clients, cleaner
+}
+
+// TearDown will delete created names using clients.
+func TearDown(clients *test.Clients, cleaner *test.Cleaner, logger *logging.BaseLogger) {
+	cleaner.Clean(true)
+
+	// There seems to be an Istio bug where if we delete / create
+	// VirtualServices too quickly we will hit pro-longed "No health
+	// upstream" causing timeouts.  Adding this small sleep to
+	// sidestep the issue.
+	//
+	// TODO(#1376):  Fix this when upstream fix is released.
```

Comment:
https://github.com/knative/serving/issues/1376 is marked as closed. I believe this is not needed anymore (and yes, knative/eventing still has it in the code).